ydb/core/scheme/scheme_tablecell.h

```diff
@@ -127,46 +196,39 @@ struct TCell {
-    }
-
-#if 1
-    // Optimization to store small values (<= 8 bytes) inplace
-    static constexpr bool CanInline(ui32 sz) { return sz <= 8; }
-    static constexpr size_t MaxInlineSize() { return 8; }
-    const char* InlineData() const                  { Y_DEBUG_ABORT_UNLESS(IsInline_); return IsNull_ ? nullptr : (char*)&IntVal; }
-    const char* Data() const                        { return IsNull_ ? nullptr : (IsInline_ ? (char*)&IntVal : Ptr); }
-#else
-    // Non-inlinable version for perf comparisons
-    static bool CanInline(ui32)                     { return false; }
-    const char* InlineData() const                  { Y_DEBUG_ABORT_UNLESS(!IsInline_); return Ptr; }
-    const char* Data() const                        { Y_DEBUG_ABORT_UNLESS(!IsInline_); return Ptr; }
-#endif
-
-    void CopyDataInto(char * dst) const {
-        if (IsInline_) {
-            switch (DataSize_) {
-                case 8: memcpy(dst, &IntVal, 8); break;
-                case 7: memcpy(dst, &IntVal, 7); break;
-                case 6: memcpy(dst, &IntVal, 6); break;
-                case 5: memcpy(dst, &IntVal, 5); break;
-                case 4: memcpy(dst, &IntVal, 4); break;
-                case 3: memcpy(dst, &IntVal, 3); break;
-                case 2: memcpy(dst, &IntVal, 2); break;
-                case 1: memcpy(dst, &IntVal, 1); break;
-            }
-            return;
-        }
+        return TCell(ptr, sizeof(val));
+    }
 
-        if (Ptr) {
-            memcpy(dst, Ptr, DataSize_);
+    void CopyDataInto(char* dst) const {
+        if (IsRefValue()) {
+            if (Ref.Size > 0) {
+                ::memcpy(dst, Ref.Ptr, Ref.Size);
+            }
+        } else {
+            // We use switch with constant size memcpy for better codegen
+            switch (Inline.Size) {
+                case 15: ::memcpy(dst, Inline.Data, 15); break;
+                case 14: ::memcpy(dst, Inline.Data, 14); break;
+                case 13: ::memcpy(dst, Inline.Data, 13); break;
+                case 12: ::memcpy(dst, Inline.Data, 12); break;
+                case 11: ::memcpy(dst, Inline.Data, 11); break;
+                case 10: ::memcpy(dst, Inline.Data, 10); break;
+                case 9: ::memcpy(dst, Inline.Data, 9); break;
+                case 8: ::memcpy(dst, Inline.Data, 8); break;
+                case 7: ::memcpy(dst, Inline.Data, 7); break;
+                case 6: ::memcpy(dst, Inline.Data, 6); break;
+                case 5: ::memcpy(dst, Inline.Data, 5); break;
+                case 4: ::memcpy(dst, Inline.Data, 4); break;
+                case 3: ::memcpy(dst, Inline.Data, 3); break;
+                case 2: ::memcpy(dst, Inline.Data, 2); break;
+                case 1: ::memcpy(dst, Inline.Data, 1); break;
+                case 0: break;
+            }
         }
     }
 };
 
-#pragma pack(pop)
-
-static_assert(sizeof(TCell) == 12, "TCell must be 12 bytes");
+static_assert(sizeof(TCell) == 16, "TCell must be 16 bytes");
```

Comment:
btw I think such comments is legacy of old standards when static_assert without macro was impossible

Reply:
True, but I don't see a point in removing the comment.